test/e2e/persistent_volumes.go

```diff
@@ -465,97 +522,256 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 		}
 	})
 
-	// Individual tests follow:
-	//
-	// Create an nfs PV, then a claim that matches the PV, and a pod that
-	// contains the claim. Verify that the PV and PVC bind correctly, and
-	// that the pod can write to the nfs volume.
-	It("should create a non-pre-bound PV and PVC: test write access [Flaky]", func() {
-
-		pv, pvc, err = createPVPVC(c, serverIP, ns, false)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
-
-		// validate PV-PVC, create and verify writer pod, delete PVC
-		// and PV
-		pv, pvc, err = completeTest(f, c, ns, pv, pvc)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
-	})
-
-	// Create a claim first, then a nfs PV that matches the claim, and a
-	// pod that contains the claim. Verify that the PV and PVC bind
-	// correctly, and that the pod can write to the nfs volume.
-	It("create a PVC and non-pre-bound PV: test write access [Flaky]", func() {
+	///////////////////////////////////////////////////////////////////////
+	//				NFS
+	///////////////////////////////////////////////////////////////////////
+	// Testing configurations of a single a PV/PVC pair, multiple evenly paired PVs/PVCs,
+	// and multiple unevenly paired PV/PVCs
+	framework.KubeDescribe("PV:NFS", func() {
 
-		pv, pvc, err = createPVCPV(c, serverIP, ns, false)
-		if err != nil {
-			framework.Failf("%v", err)
+		pvConfig = persistentVolumeConfig {
+			pvsource: api.PersistentVolumeSource{
+				NFS: &api.NFSVolumeSource{
+					Server:   serverIP,
+					Path:     "/exports",
+					ReadOnly: false,
+				},
+			},
 		}
 
-		// validate PV-PVC, create and verify writer pod, delete PVC
-		// and PV
-		pv, pvc, err = completeTest(f, c, ns, pv, pvc)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
-	})
+		Context("with Single PV - PVC pairs", func() {
 
-	// Create a claim first, then a pre-bound nfs PV that matches the claim,
-	// and a pod that contains the claim. Verify that the PV and PVC bind
-	// correctly, and that the pod can write to the nfs volume.
-	It("create a PVC and a pre-bound PV: test write access [Flaky]", func() {
+			var pv *api.PersistentVolume
+			var pvc *api.PersistentVolumeClaim
 
-		pv, pvc, err = createPVCPV(c, serverIP, ns, true)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
+			pvConfig = persistentVolumeConfig{
```

Comment:
Why is pvConfig set here in in KubeDescribe() above? It appears to be the same definition.